lib/charms/mongodb_libs/v0/relations.py

```diff
@@ -0,0 +1,171 @@
+            event.defer()
+            return
+
+    def reconcile(self, departed_relation_id: Optional[int]):
+        """Forced reconciliation of all current relations."""
+        with MongoDBConnection(self.charm.mongodb_config) as mongo:
+            database_users = mongo.get_users()
+            relation_users = self._get_users_from_relations(departed_relation_id)
+
+            for username in database_users - relation_users:
+                logger.info("Remove relation user: %s", username)
+                mongo.drop_user(username)
+
+            for username in relation_users - database_users:
+                config = self._get_config(username)
+                if config.database is None:
+                    # We need to wait for moment when provider library
+                    # set the database name into the relation.
+                    continue
+                logger.info("Create relation user: %s on %s", config.username, config.database)
+                mongo.create_user(config)
+                self._set_relation(config)
+
+            for username in relation_users.intersection(database_users):
+                config = self._get_config(username)
+                logger.info("Update relation user: %s on %s", config.username, config.database)
+                mongo.update_user(config)
+
+            if not self.charm.model.config["auto-delete"]:
+                return
+
+            database_dbs = mongo.get_databases()
+            relation_dbs = self._get_databases_from_relations(departed_relation_id)
+            for database in database_dbs - relation_dbs:
+                logger.info("Drop database: %s", database)
+                mongo.drop_database(database)
+
+    def _get_config(self, username: str) -> MongoDBConfiguration:
+        """Construct config object for future user creation."""
+        relation = self._get_relation(username)
+        return MongoDBConfiguration(
+            replset=self.charm.app.name,
+            database=self._get_relation_database(relation),
+            username=username,
+            password=generate_password(),
+            hosts=self.charm.mongodb_config.hosts,
+            roles=self._get_relation_roles(relation),
+        )
+
+    def _set_relation(self, config: MongoDBConfiguration):
+        """Save all output fields into application relation."""
+        relation = self._get_relation(config.username)
+        if relation is None:
+            return None
+        relation.data[self.charm.app]["username"] = config.username
+        relation.data[self.charm.app]["password"] = config.password
+        relation.data[self.charm.app]["database"] = config.password
+        relation.data[self.charm.app]["endpoints"] = ",".join(config.hosts)
+        relation.data[self.charm.app]["replset"] = config.replset
+        relation.data[self.charm.app]["uris"] = config.uri
+
+    @staticmethod
+    def _get_username(relation_id: str) -> str:
```

Comment:
```suggestion
    def _get_username_from_relation(relation_id: str) -> str:
```
follow naming scheme

Reply:
fixed in https://github.com/delgod/mongodb-k8s-operator/pull/3/commits/d4e3681756f12ba7282b634ae47627079c74b0cc